private-templates-service/client/src/components/Designer/Designer.tsx

```diff
@@ -60,18 +66,43 @@ interface State {
   isSaveOpen: boolean;
 }
 
+interface MatchParams {
+  uuid: string;
+  version: string;
+}
+
 let designer: ACDesigner.CardDesigner;
 
 class Designer extends React.Component<DesignerProps,State> {
   constructor(props: DesignerProps) {
     super(props);
     props.setPage(this.props.templateName, "Designer");
     this.state = {isSaveOpen: false };
+    if(this.props.match.params.uuid !== "newcard"){
+      this.props.getTemplate(this.props.match.params.uuid);
+    }
   }
 
   toggleModal = () => {
     this.setState({isSaveOpen: !this.state.isSaveOpen});
   }
+  
+  updateURL = () => {
+      this.props.history.push("/designer/"+ this.props.templateID + "/" + this.props.version);
+  }
+  
+  toggleSpinner = (isFetching: boolean) => {
+    if(isFetching){
+      this.props.openModal(ModalState.Spinner)
+    }
+  }
```

Comment:
Why do we do this:
`isFetching === true -> component did update -> dispatch action to open spinner modal -> ModalState === spinner -> spinner open`

Why cant we do this
`isFetching === true -> spinner open`
By extension, `isFetching === false -> spinner not open`